### ISSUE TITLE
Builder#parse_env does not override manually set filter level

### ISSUE THREAD

**Issue by BramBonne** [env-logger-rs/env_logger#196]
Consider the following example ([runnable version](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=db7652ef4104ae10fbdadc97b4c74ce6)):
```rust
std::env::set_var("RUST_LOG", "debug");

let mut builder = env_logger::Builder::new();
builder.filter_level(log::LevelFilter::Trace);

// The following call should override the log level to Debug, as read from the environment
builder.parse_default_env();
    
builder.init();
assert_eq!(log::max_level(), log::LevelFilter::Trace);
```

According to the [documentation](https://docs.rs/env_logger/0.8.3/env_logger/struct.Builder.html#method.parse_env), the `parse_env` function "allows a builder to be configured with default parameters, to be then overridden by the environment". For this example, that would mean that `log::max_level()` should be set to `Debug` at the end of this example. In reality, it's set to `Trace` instead.

Removing the call to `builder.filter_level(log::LevelFilter::Trace)` correctly sets the log level to `Debug` instead.

**Comment by jplatte** [comment]
Yeah, the docs are pretty clear here. A PR fixing this would be welcome.

**Comment by BramBonne** [comment]
I provided both #199 and #200 as alternative potential fixes to this problem, depending on whether we want directive names to be unique in general (#200), or not (#199).

**Comment by BramBonne** [comment]
Hey @jplatte, friendly ping on this. Would #200 be an acceptable way to fix this, or is there a better approach? Thanks for taking a look :-)

Kind regards,
Bram

**Comment by SirWindfield** [comment]
I'll have time on the weekend to take a look at the PR.

**Comment by BramBonne** [comment]
Thank you! But no rush, I was just checking it was still being considered 🙂

**Comment by SirWindfield** [comment]
@jplatte I've taken a look at both of the PRs. Both look good.

I don't have strong preference for either PR, but if I'd have to choose I'd say we should merge #200, as it means more cosistent behaviour across the library in general by __always__ overwriting previous directives. I can't think of an example off the top of my head where this might not be wanted.

**Comment by jplatte** [comment]
I've had a look, I agree that #200 seems like the better solution.

**Comment by BramBonne** [comment]
Hey,

Just checking if anything is expected/needed from my end here.

Kind regards,
Bram

**Comment by jplatte** [comment]
No. I merged #200. Sorry about the long delay.
